bundles/org.corpus_tools.hexatomic.graph/src/main/java/org/corpus_tools/hexatomic/graph/GraphEditor.java

```diff
@@ -0,0 +1,713 @@
+
+  @Inject
+  private MPart thisPart;
+
+  @Inject
+  public GraphEditor() {
+
+  }
+
+  private Button btnIncludeSpans;
+  private Table textRangeTable;
+  private Text txtSegmentFilter;
+  private Button btnIncludePointingRelations;
+
+  private GraphViewer viewer;
+
+
+  @Inject
+  UISynchronize sync;
+
+  @Inject
+  ErrorService errors;
+
+  private ListenerImplementation projectChangeListener;
+
+  private SDocumentGraph getGraph() {
+    String documentID = thisPart.getPersistedState().get("org.corpus_tools.hexatomic.document-id");
+    Optional<SDocument> doc = projectManager.getDocument(documentID);
+    if (doc.isPresent()) {
+      return doc.get().getDocumentGraph();
+    }
+    return null;
+  }
+
+  /**
+   * Create a new graph viewer.
+   * 
+   * @param parent The parent SWT composite.
+   * @param part The part this viewer belongs to.
+   */
+  @PostConstruct
+  public void postConstruct(Composite parent, MPart part) {
+
+    projectChangeListener = new ListenerImplementation();
+    projectManager.addListener(projectChangeListener);
+
+    parent.setLayout(new FillLayout(SWT.VERTICAL));
+
+    SashForm graphSash = new SashForm(parent, SWT.HORIZONTAL);
+
+    viewer = new GraphViewer(graphSash, SWT.BORDER | SWT.H_SCROLL | SWT.V_SCROLL);
+    viewer.getGraphControl().setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true, 1, 1));
+    viewer.setContentProvider(new SaltGraphContentProvider());
+    viewer.setLabelProvider(
+        new SaltGraphStyler(viewer.getGraphControl().getLightweightSystem().getRootFigure()));
+    viewer.setLayoutAlgorithm(createLayout());
+    viewer.setNodeStyle(ZestStyles.NODES_NO_LAYOUT_ANIMATION);
+    viewer.setConnectionStyle(ZestStyles.CONNECTIONS_DIRECTED);
+    viewer.getGraphControl().setDragDetect(true);
+
+    Composite filterComposite = new Composite(graphSash, SWT.NONE);
+    GridLayout gridLayoutFilterComposite = new GridLayout(1, false);
+    gridLayoutFilterComposite.marginWidth = 0;
+    filterComposite.setLayout(gridLayoutFilterComposite);
+
+    Label lblFilterByAnnotation = new Label(filterComposite, SWT.NONE);
+    lblFilterByAnnotation.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false, 1, 1));
+    lblFilterByAnnotation.setText("Filter by annotation type");
+
+    btnIncludeSpans = new Button(filterComposite, SWT.CHECK);
+    btnIncludeSpans.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false, 1, 1));
+    btnIncludeSpans.setSelection(false);
+    btnIncludeSpans.setText("Include Spans");
+
+    btnIncludePointingRelations = new Button(filterComposite, SWT.CHECK);
+    btnIncludePointingRelations.setLayoutData(new GridData(SWT.LEFT, SWT.TOP, true, false, 1, 1));
+    btnIncludePointingRelations.setSelection(true);
+    btnIncludePointingRelations.setText("Include pointing relations");
+
+    txtSegmentFilter = new Text(filterComposite, SWT.BORDER);
+    txtSegmentFilter.setLayoutData(new GridData(SWT.LEFT, SWT.TOP, true, false, 1, 1));
```

Comment:
```suggestion
    txtSegmentFilter.setMessage("Filter by node annotation name");
    txtSegmentFilter.setLayoutData(new GridData(SWT.FILL, SWT.TOP, true, false, 1, 1));
```

Set a message to explain what this field does, and make the field fill the horizontal space.